### PR TITLE
updating cache merge strategy to support entity removal

### PR DESCRIPTION
**Pull request by SomkaPe**

### CODE REVIEW

File: src/main/java/com/microsoft/aad/msal4j/TokenCache.java [hunk 1]
```diff
@@ -103,6 +110,25 @@ private static void mergeJsonObjects(JsonObject old, JsonObject update) {
         }
     }
 
+    private static void mergeRemovals(JsonObject old, JsonObject update, String parentKey) {
+        Set<String> msalEntities =
+                new HashSet<>(Arrays.asList("Account", "sds", "AccessToken", "RefreshToken", "IdToken", "AppMetadata"));
```
Comment (henrik-me) [inline src/main/java/com/microsoft/aad/msal4j/TokenCache.java]: >sds [](start = 56, length = 3)

sorry for my ignorance. what is this?

File: src/main/java/com/microsoft/aad/msal4j/TokenCache.java [hunk 2]
```diff
@@ -103,6 +110,25 @@ private static void mergeJsonObjects(JsonObject old, JsonObject update) {
         }
     }
 
+    private static void mergeRemovals(JsonObject old, JsonObject update, String parentKey) {
```
Comment (henrik-me) [inline src/main/java/com/microsoft/aad/msal4j/TokenCache.java]: I'm not seeing this being used? am I missing something?

Comment (SomkaPe) [reply src/main/java/com/microsoft/aad/msal4j/TokenCache.java]: it is used in mergeJsonObjects function

Comment (henrik-me) [reply src/main/java/com/microsoft/aad/msal4j/TokenCache.java]: where?

File: src/main/java/com/microsoft/aad/msal4j/TokenCache.java (hunk 1 above)
Comment (henrik-me) [reply src/main/java/com/microsoft/aad/msal4j/TokenCache.java]: what is sds?

File: src/main/java/com/microsoft/aad/msal4j/TokenCache.java (hunk 2 above)
Comment (SomkaPe) [reply src/main/java/com/microsoft/aad/msal4j/TokenCache.java]: removed

**Pull request merged**